src/main/scala/ai/privado/tagger/sink/SinkArgumentUtility.scala

```diff
@@ -0,0 +1,333 @@
+package ai.privado.tagger.sink
+
+import io.shiftleft.codepropertygraph.generated.nodes.{
+  AstNode,
+  CfgNode,
+  NewFile,
+  NewTag,
+  Block,
+  Literal,
+  Call,
+  Identifier,
+  FieldIdentifier,
+  Local
+}
+import io.shiftleft.codepropertygraph.generated.{EdgeTypes, Operators}
+import io.shiftleft.semanticcpg.language._
+import io.shiftleft.utils.IOUtils
+import io.joern.dataflowengineoss.DefaultSemantics
+import io.joern.dataflowengineoss.queryengine.{EngineConfig, EngineContext}
+import io.joern.dataflowengineoss.semanticsloader.Semantics
+import io.shiftleft.codepropertygraph.generated.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes.JavaProperty
+
+import scala.collection.mutable.ListBuffer
+import scala.util.matching.Regex
+import scala.collection.mutable.ListBuffer
+import java.io.{PrintWriter, ByteArrayInputStream, ByteArrayOutputStream, ObjectInputStream, ObjectOutputStream}
+import java.nio.file.Files
+
+object SinkArgumentUtility {
+
+  /** Populates argumentList for a GAPixel node. This function processes the given `node` to extract the arguments it
+    * contains. Arguments can be literals, identifiers, or derived from assignment, call Nodes and spread operator
+    * nodes.
+    *
+    * @param node
+    *   The Call node representing the GAPixel.
+    * @param cpg
+    *   The Code Property Graph instance.
+    * @return
+    *   A list of key-value pairs representing the extracted arguments.
+    */
+  def addArgumentsForGAPixelNode(node: Call, cpg: Cpg): List[(String, String)] = {
+    val keyValueStructures = ListBuffer.empty[(String, String)]
+
+    val assignmentNodes     = node.argument.isBlock.astChildren.isCall.name(Operators.assignment).l
+    val spreadOperatorNodes = node.argument.isBlock.astChildren.isCall.name("<operator>.spread").l
+    val identifierNodes     = node.argument.isIdentifier.l
+    val literalNodes        = node.argument.isLiteral.l
+
+    literalNodes.foreach { l =>
+      keyValueStructures += ((l.code, l.code))
+    }
+    identifierNodes.foreach { i =>
+      processIdentifierNode(i, "", cpg, keyValueStructures)
+    }
+
+    processAssignmentNodes(assignmentNodes, "", cpg, keyValueStructures)
+    processSpreadOperatorNodes(spreadOperatorNodes, "", cpg, keyValueStructures)
+
+    keyValueStructures.toList
+  }
+
+  /** Processes an Identifier node to extract its type information and member names. If typeDecl found for identifier,
+    * check for members related to it.
+    */
+  private def processIdentifierNode(
+    n: Identifier,
+    topLeftKey: String,
+    cpg: Cpg,
+    keyValueStructures: ListBuffer[(String, String)]
+  ): Unit = {
+    def updateLeftKey(code: String): String = {
+      if (topLeftKey.nonEmpty)
+        topLeftKey + "." + code
+      else
+        code
+    }
```

Comment:
[Optional] - This can be written as a expression
```
def updateLeftKey(code: String): String = if topLeftKey.nonEmpty then s"$topLeftKey.$code" else code